modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java

```diff
@@ -795,20 +788,11 @@ private GridCloseableIterator<IgniteBiTuple<K, V>> sharedCacheSetIterator(CacheQ
     private GridCloseableIterator scanIterator(final CacheQuery<?> qry, IgniteClosure transformer,
         boolean locNode)
         throws IgniteCheckedException {
-        final InternalScanFilter<K, V> intFilter = internalFilter(qry.scanFilter());
-
         try {
             Integer part = qry.partition();
 
             if (part != null && (part < 0 || part >= cctx.affinity().partitions()))
-                return new GridEmptyCloseableIterator() {
-                    @Override public void close() throws IgniteCheckedException {
-                        if (intFilter != null)
```

Comment:
Why we don't need to close qry.scanFilter() anymore?

Reply:
Because, here we don't allocate any resources.
It required and happens only inside `ScanQueryIterator`.

`ScanQueryIterator` correctly manage those resources inside.